snapcraft/plugins/colcon.py

```diff
@@ -0,0 +1,687 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The colcon plugin is useful for building ROS2 parts.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - colcon-packages:
+      (list of strings)
+      List of colcon packages to build. If not specified, all packages in the
+      workspace will be built. If set to an empty list ([]), no packages will
+      be built.
+    - colcon-source-space:
+      (string)
+      The source space containing colcon packages (defaults to 'src').
+    - colcon-rosdistro:
+      (string)
+      The ROS distro to use. Available options are bouncy and crystal (defaults to
```

Comment:
Unlike ROS1 which has settled into a single release per Ubuntu LTS, ROS2 cranks out multiple releases for the same LTS, which means we can't use the base to determine the ROS distro.

Reply:
Yeah, sorry, I was trying to avoid being imperative. I was trying to get to the point that the docs reflect that the choice of ROS release is not available to all bases (and maybe even specify that the default assumes/requires core18)

Reply:
Updated the docs, how does that look?